### DISCUSSION TITLE
Compilation error after update to ESP32 library to 3.0.1

### DISCUSSION THREAD

**Discussion by gmoccapy** [Bodmer/TFT_eSPI#3361] [Alerts]
Hallo, 
hopping I can help to avoid errors to new users, I would like to report:

I just updated the ESP32 Board Manager in Arduino IDE from release 2.0.17 to 3.0.1 and without changing any code I get now an compilation error:

`In file included from /home/gmoccapy/Arduino/libraries/TFT_eSPI/TFT_eSPI.cpp:24:
/home/gmoccapy/Arduino/libraries/TFT_eSPI/Processors/TFT_eSPI_ESP32.c: In member function 'uint8_t TFT_eSPI::readByte()':
/home/gmoccapy/Arduino/libraries/TFT_eSPI/Processors/TFT_eSPI_ESP32.c:113:9: error: 'gpio_input_get' was not declared in this scope; did you mean 'gpio_num_t'?
  113 |   reg = gpio_input_get(); // Read three times to allow for bus access time
      |         ^~~~~~~~~~~~~~
      |         gpio_num_t

exit status 1

Compilation error: exit status 1`

After downgrading again to 2.0.17 everythink is working as before.

Regards Norbert

**Comment by Bodmer** [comment]
Unfortunately there appears to be no stable API for the Espressif processors and the software sturcture and naming keeps changing so it is difficult and time consuming to keep the library compatible with each ESP32 board package update cycle.

I don't have time to look at this and fix it so using 2.0.17 is the solution at the moment.

**Reply by Cincinnatu** [reply]
Can you tell me how to change it? Because I now have to work on the 3.0 version。

**Comment by gkeepleft** [comment]
@Bodmer thanks for the quick solution. I downloaded version 3 on a big project and was about to give up on TFT_eSPI when i saw your reply. Thanks mate, you helped a lot with that comment. It would be nice if Espressif could be a tad more considerate of such an awesome and expansive library. Dont get me wrong, I'm in awe of the ESP line up, but they should consider it more of a symbiotic relationship with people like you. TFT_eSPI is incredibly powerful and I view it as a rare treat in this world.
